src/package_manager.rs

```diff
@@ -29,6 +30,85 @@ impl From<&OperatingSystem> for PackageManager {
     }
 }
 
+impl PackageManager {
+    /// Install a package using the package manager.
+    pub fn install(&self, instruction: &Install) -> DepotResult<()> {
+        match self {
+            PackageManager::Pacman => println!("pacman -S {}", instruction.package.join(" ")),
+            PackageManager::Yay => println!("yay -S {}", instruction.package.join(" ")),
+            PackageManager::Apk => println!("apk add {}", instruction.package.join(" ")),
+            PackageManager::AptGet => println!("apt-get install {}", instruction.package.join(" ")),
+            PackageManager::Apt => println!("apt install {}", instruction.package.join(" ")),
+            PackageManager::Pkg => println!("pkg install {}", instruction.package.join(" ")),
+            PackageManager::Dnf => println!("dnf install {}", instruction.package.join(" ")),
+        };
+        Ok(())
```

Comment:
I know this may seem useless but having these methods return a Result will allow us to implement the commands without touching the main.rs